backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/VersionInputHandler.java

```diff
@@ -0,0 +1,245 @@
+                                .bind("entity_id", entityId)
+                                .execute();
+                          }
+                          return rs;
+                        })
+                    .list();
+              }
+            }
+          }
+        });
+  }
+
+  /**
+   * @param versioningEntry : versioningEntry
+   * @return returns a map from location to an Entry of BlobExpanded and sha
+   * @throws ModelDBException ModelDBException
+   */
+  private InternalFuture<Map<String, Map.Entry<BlobExpanded, String>>> validateVersioningEntity(
+      VersioningEntry versioningEntry) {
+    String errorMessage = null;
+    if (versioningEntry.getRepositoryId() == 0L) {
+      errorMessage = "Repository Id not found in VersioningEntry";
+    } else if (versioningEntry.getCommit().isEmpty()) {
+      errorMessage = "Commit hash not found in VersioningEntry";
+    }
+
+    if (errorMessage != null) {
+      throw new ModelDBException(errorMessage, io.grpc.Status.Code.INVALID_ARGUMENT);
+    }
+    Map<String, Map.Entry<BlobExpanded, String>> requestedLocationBlobWithHashMap = new HashMap<>();
+    try (Session session = modelDBHibernateUtil.getSessionFactory().openSession()) {
+      RepositoryIdentification repositoryIdentification =
+          RepositoryIdentification.newBuilder()
+              .setRepoId(versioningEntry.getRepositoryId())
+              .build();
+      CommitEntity commitEntity =
+          commitDAO.getCommitEntity(
+              session,
+              versioningEntry.getCommit(),
+              (session1) -> repositoryDAO.getRepositoryById(session, repositoryIdentification));
+      if (!versioningEntry.getKeyLocationMapMap().isEmpty()) {
+        Map<String, Map.Entry<BlobExpanded, String>> locationBlobWithHashMap =
+            blobDAO.getCommitBlobMapWithHash(
+                session, commitEntity.getRootSha(), new ArrayList<>(), Collections.emptyList());
+        for (Map.Entry<String, Location> locationBlobKeyMap :
+            versioningEntry.getKeyLocationMapMap().entrySet()) {
+          String locationKey = String.join("#", locationBlobKeyMap.getValue().getLocationList());
+          if (!locationBlobWithHashMap.containsKey(locationKey)) {
+            throw new ModelDBException(
+                "Blob Location '"
+                    + locationBlobKeyMap.getValue().getLocationList()
+                    + "' for key '"
+                    + locationBlobKeyMap.getKey()
+                    + "' not found in commit blobs",
+                io.grpc.Status.Code.INVALID_ARGUMENT);
+          }
+          requestedLocationBlobWithHashMap.put(
+              locationKey, locationBlobWithHashMap.get(locationKey));
+        }
+      }
+    } catch (ModelDBException ex) {
+      throw ex;
+    } catch (Exception ex) {
+      throw new ModelDBException(ex);
+    }
+    return InternalFuture.completedInternalFuture(requestedLocationBlobWithHashMap);
```

Comment:
This value is always known immediately, so remove the future wrapper for clarity.

Reply:
Handled using InternalFuture Void